### DISCUSSION TITLE
Implementing my own LLM API

### DISCUSSION THREAD

**Discussion by westlongtime** [Holmeswww/AgentKit#21] [Q&A] [answered]
how to bulid AgentKit without using api?  I want to load my local LLM

**Comment by Holmeswww** [comment, marked as answer]
`LLM_API_FUNCTION` can be any LLM API function that takes `msg:list` and `shrink_idx:int`, and outputs `llm_result:str` and `usage:dict`. Where msg is a prompt ([OpenAI format](https://platform.openai.com/docs/guides/text-generation/chat-completions-api) by default), and `shrink_idx:int` is an index at which the LLM should reduce the length of the prompt in case of overflow.

You can write your custom LLM function by extending or following the format of https://github.com/Holmeswww/AgentKit/blob/main/src/agentkit/llm_api/base.py
